File: 0101-0200/0167.Two_Sum_II-Input_Array_Is_Sorted.cpp
//0ms





class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int n=numbers.size();
        int i=0,j=n-1,sum=0;
        while(i<j){
            sum=numbers[i]+numbers[j];
            if(sum==target) return {i+1, j+1};
            else if(sum<target){
            i ++;}
            else{
            j -- ;}
            }
        
      return {};
    }
};





//0ms





class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int index1 = 0; 
        int index2 = numbers.size()-1;
        int sum = numbers[index1] + numbers[index2];
        while (sum != target) {
            if (sum > target) {
                index2--;
            } else {
                index1++;
            }
            sum = numbers[index1] + numbers[index2];
        }
        return {index1+1, index2+1};
    }
};

File: 0101-0200/0167.Two_Sum_II-Input_Array_Is_Sorted.rs
//0ms






use std::cmp::Ordering;

impl Solution {
    pub fn two_sum(numbers: Vec<i32>, target: i32) -> Vec<i32> {
        let mut i = 0;
        let mut j = numbers.len()-1;
        
        while i < j {
            match (numbers[i]+numbers[j]).cmp(&target) {
                Ordering::Less => i += 1,
                Ordering::Greater => j -= 1,
                Ordering::Equal => return vec![(i+1) as i32, (j+1) as i32],
            }
        }
        unreachable!();
    }
}







0ms






impl Solution {
    pub fn two_sum(numbers: Vec<i32>, target: i32) -> Vec<i32> {
        let (mut right, mut left) = (numbers.len().saturating_sub(1), 0);
        while numbers[left] + numbers[right] != target {
            if numbers[left] + numbers[right] > target {
                right -= 1;
            } else {
                left += 1;
            }
        }
        return vec![(left + 1) as i32, (right + 1) as i32];
    }
}
